packages/gatsby-image/src/index.js

```diff
@@ -516,7 +565,7 @@ class Image extends React.Component {
 
     if (fixed) {
       const imageVariants = fixed
-      const image = imageVariants[0]
+      const image = getCurrentSrcData({ fixed })
```

Comment:
Like the other call, pass on `fixed` as an argument. They have the same structure anyways.